mdaviz/mda_file.py

```diff
@@ -0,0 +1,589 @@
+        scanDict, first_pos, first_det = utils.get_scan(file_data)
+        pvList = [v["name"] for v in scanDict.values()]
+        self._data = {
+            "fileName": file_path.stem,  # file_name.rsplit(".mda", 1)[0]
+            "filePath": str(file_path),
+            "folderPath": str(folder_path),
+            "metadata": file_metadata,
+            "scanDict": scanDict,
+            "firstPos": first_pos,
+            "firstDet": first_det,
+            "pvList": pvList,
+            "index": index,
+        }
+
+    def setStatus(self, text):
+        self.mda_mvc.setStatus(text)
+
+    # ------ Tab utilities:
+
+    def tabPath2Index(self, file_path):
+        """Finds and returns the index of a tab based on its associated file path."""
+        if file_path is not None:
+            for tab_index in range(self.tabWidget.count()):
+                tab_tableview = self.tabWidget.widget(tab_index)
+                if tab_tableview.filePath.text() == file_path:
+                    return tab_index
+        return None  # Return None if the file_path is not found.
+
+    def tabIndex2Path(self, index):
+        """Returns the file path associated with a given tab index."""
+        if 0 <= index < self.tabWidget.count():
+            tab_tableview = self.tabWidget.widget(index)
+            return tab_tableview.filePath.text()
+        return None  # Return None if the index is out of range.
+
+    def tabPath2Tableview(self, file_path):
+        """Finds and returns the tableview of a tab based on its associated file path."""
+        if file_path is not None:
+            for tab_index in range(self.tabWidget.count()):
+                tab_tableview = self.tabWidget.widget(tab_index)
+                if tab_tableview.filePath.text() == file_path:
+                    return tab_tableview
+        return None  # Return None if the file_path is not found.
+
+    def tabIndex2Tableview(self, index):
+        """Returns the Tableview associated with a given tab index."""
+        if 0 <= index < self.tabWidget.count():
+            tab_tableview = self.tabWidget.widget(index)
+            return tab_tableview
+        return None  # Return None if the index is out of range.
+
+    # ------ Populating UIs with selected file content:
+
+    def displayMetadata(self, metadata):
+        """Display metadata in the vizualization panel."""
+        if not metadata:
+            return
+        metadata = utils.get_md(metadata)
+        metadata = yaml.dump(metadata, default_flow_style=False)
+        self.mda_mvc.mda_file_viz.setMetadata(metadata)
+
+    def displayData(self, tabledata):
+        """Display pos(s) & det(s) values as a tableview in the vizualization panel."""
+        if not self.data():
+            return
+        self.mda_mvc.mda_file_viz.setTableData(tabledata)
+
+    def defaultSelection(self, first_pos, first_det, selection_field):
+        """Sets the default field selection if no selection is provided.
+        Args:
```

Comment:
Needs a blank line before this one.  Formatting such as this will become apparent once we build documentation from the source code.